### PR TITLE
Use specific ruby thread to handle auth metadata plugin callbacks

### LINKED ISSUE #4598: Ruby auth plugin test failure
https://grpc-testing.appspot.com/job/gRPC_master/14209/config=opt,language=ruby,platform=linux/testReport/junit/(root)/tests/tools_run_tests_run_ruby_sh/

```
[BUG] rb_thread_call_with_gvl() is called by non-ruby thread
```

This is most likely a bug in the auth metadata plugin.

### PR DESCRIPTION
**Pull request by murgatroid99**
This should fix #4598.

### CODE REVIEW

File: src/ruby/ext/grpc/rb_event_thread.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  event->argument = argument;
+  event->next = NULL;
+  gpr_mu_lock(event_queue.mu);
+  if (event_queue.tail == NULL) {
+    event_queue.head = event_queue.tail = event;
+  } else {
+    event_queue.tail->next = event;
+    event_queue.tail = event;
+  }
+  gpr_mu_unlock(event_queue.mu);
```
Comment (ctiller) [inline src/ruby/ext/grpc/rb_event_thread.c]: It's better to signal() then unlock()

File: src/ruby/ext/grpc/rb_event_thread.c [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  }
+  gpr_mu_unlock(event_queue.mu);
+  return event;
+}
+
+static void grpc_rb_event_unblocking_func(void *arg) {
+  gpr_mu_lock(event_queue.mu);
+  event_queue.abort = true;
+  gpr_mu_unlock(event_queue.mu);
```
Comment (ctiller) [inline src/ruby/ext/grpc/rb_event_thread.c]: here too

File: src/ruby/ext/grpc/rb_event_thread.c [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      gpr_free(event);
+    }
+  }
+  grpc_rb_event_queue_destroy();
+  return Qnil;
+}
+
+void grpc_rb_event_queue_thread_start() {
+
+  event_queue.mu = gpr_malloc(sizeof(gpr_mu));
```
Comment (ctiller) [inline src/ruby/ext/grpc/rb_event_thread.c]: No need to allocate these, just put a gpr_mu in the struct inline.

gpr_mu_init(&event_queue.mu);

File: src/ruby/ext/grpc/rb_event_thread.c [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  event_queue.cv = gpr_malloc(sizeof(gpr_cv));
```
Comment (ctiller) [inline src/ruby/ext/grpc/rb_event_thread.c]: same here

**Comment by murgatroid99** [comment]
Addressed those comments

**Pull request merged**